Features/BanCommand.cs

```diff
@@ -0,0 +1,89 @@
+using DSharpPlus.Commands;
+using DSharpPlus.Entities;
+using DSharpPlus.Commands.ContextChecks;
+using ArtcordAdminBot.Features.Helpers;
+using System.IO;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+
+namespace ArtcordAdminBot.Features
+{
+    public class BanCommand
+    {
+        private readonly IDatabaseService _databaseService;
+        private readonly HttpClient _httpClient;
+
+        public BanCommand(IDatabaseService databaseService)
+        {
+            _databaseService = databaseService;
+            _httpClient = new HttpClient();
+        }
+
+        [Command("ban")]
+        [RequirePermissions(DiscordPermissions.BanMembers)]
+        public async Task BanAsync(CommandContext ctx, 
+        [System.ComponentModel.Description("The user to ban.")]DiscordUser targetUser, 
+        [System.ComponentModel.Description("The reason for the ban.")] string reason = "No reason provided", 
```

Comment:
I would make this `reason` variable nullable (and default null), so if there is no reason provided it will actually be stored as no reason (`null`) in the database. 

The only time when you should replace this with the string `"No reason provided"` is when printing the reason out. Only then replace the `null` with a default value.